src/statemachine/blender.js

```diff
@@ -0,0 +1,250 @@
+                if (t >= 1) {
+                    return false;
+                }
+            }, undefined, bind);
+        };
+    }
+
+    //Find the difference between two quaternions
+    function diffQuat(q2, q1) {
+        var a = q1.clone().invert();
+        return a.mul(q2);//
+    }
+
+
+
+    function value(start, end, t) {
+        var result = start;
+        if (start instanceof pc.Vec3) {
+            result = end.clone().sub(start).scale(t).add(start);
+        }
+        else if (start instanceof pc.Quat) {
+            result = new pc.Quat().slerp(pc.Quat.IDENTITY, diffQuat(end, start), t).mul(start);
+        }
+        else if (!isNaN(start)) {
+            result = (+end - +start) * t + +start;
+        }
+        return result;
+    }
+
+    function lerp(start, end, t) {
+        return value(start, end, pc.math.clamp(t, 0, 1));
+    }
+
+    function smooth(start, end, t) {
+        return interpolate.value(start, end, pc.math.smootherstep(0, 1, t));
+    }
+
+    /**
+     * @callback
+     * @name pc.interpolate~valuecb
+     * @description call back containing the value of an interpolation
+     * @param {pc.Vec3|pc.Quat|Number} value The current value
+     * @param {Number} t The current proportion of the interpolation that is complete 0..1
+     */
+
+    var interpolate = {
+        /**
+         * @name pc.interpolate.value
+         * @function
+         * @description Get a value between start and end with unconstrained t (can be > 1 or < 0)
+         * @param {pc.Vec3|pc.Quat|Number} start The start point for the interpolation
+         * @param {pc.Vec3|pc.Quat|Number}end The end point for the interpolation
+         * @param {Number} t The position between start and end
+         * @returns {pc.Vec3|pc.Quat|Number} The point at the specified time
+         */
+        value: value,
+        /**
+         * @name pc.interpolate.lerp
+         * @function
+         * @description lerp a value between start and end with constrained t (forced between 0 and
+         *     1)
+         * @param {pc.Vec3|pc.Quat|Number} start The start point for the interpolation
+         * @param {pc.Vec3|pc.Quat|Number} end The end point for the interpolation
+         * @param {Number} t The position between start and end
+         * @returns {pc.Vec3|pc.Quat|Number} The lerped value
+         */
+        lerp: lerp,
+        /**
+         * @name pc.interpolate.smooth
+         * @function
+         * @description Use smoothstep to make a natural lerp of a value between start and end with
+         *     constrained t (forced between 0 and 1)
+         * @param {pc.Vec3|pc.Quat|Number} start The start point for the interpolation
+         * @param {pc.Vec3|pc.Quat|Number} end The end point for the interpolation
+         * @param {Number} t The position between start and end
+         * @returns {pc.Vec3|pc.Quat|Number} The smoothly interpolated value
+         */
+        smooth: smooth,
```

Comment:
I guess this would be a good place to add more types of interpolation in the future? Like easeIn easeOut etc?